tests/Feature/Api/Validation/AddressTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Feature\Api\Validation;
+
+use GuzzleHttp\Exception\ClientException;
+use jsamhall\ShipEngine\Address\Address;
+use PHPUnit\Framework\TestCase;
+use Tests\Mocks\Validation\AddressMock;
+
+class AddressTest extends TestCase
+{
+
+    public function testValidateAddress()
+    {
+        //Arrange
+        $address = new Address();
+        $address->setAddressLine1("525 S Winchester Blvd")
+            ->setCityLocality("San Jose")
+            ->setStateProvince("CA")
+            ->setPostalCode("95128")
+            ->setCountryCode("US")
+            ->setAddressResidentialIndicator("unknown");
+        $shipEngineMock = new AddressMock();
+
+        $request = $shipEngineMock->mockValidationSuccess();
+
+        //Act
+        $response = $request->validateAddress($address);
+
+        //Assert
+        $this->assertSame(strtoupper($address->getAddressLine1()), $response->getMatchedAddress()->getAddressLine1());
+        $this->assertSame(strtoupper($address->getCityLocality()), $response->getMatchedAddress()->getCityLocality());
+    }
+
+    public function testInsufficientValidation()
```

Comment:
The tests can return `:void` as well

Reply:
Just don't stare into it. But yeah, it's done.